api/controller/dbController.test.js

```diff
@@ -0,0 +1,96 @@
+      const artistId = await db.getArtist("nirvana")
+      expect(await db.getDates({artistId,songName:"lithium"})).toEqual(sampleResult);
+    });
+    it('tests the getArtist function to return proper values', async () => {
+      const sampleApiReturn = await readFile('./sampleReturn.json', 'utf8');
+      const apiReturn = JSON.parse(sampleApiReturn);
+      const db = new dbController(apiReturn,"nirvana","lithium",1234)
+      
+      expect(await db.getArtist("nirvana")).toEqual(54);
+    });
+    it('tests the getSong function to return proper values', async () => {
+      const sampleApiReturn = await readFile('./sampleReturn.json', 'utf8');
+      const apiReturn = JSON.parse(sampleApiReturn);
+      const db = new dbController(apiReturn,"nirvana","lithium",1234)
+      expect(await db.getSong({songName:"lithium",artistId:54})).toEqual(5161);
+    });
+    it('mocks the addData function', async () => {
```

Comment:
So let's keep this test for now, especially as we migrate it to another file and pull it apart. However I think that in the future we can test this differently. Instead of checking how many times certain methods have been called we can see how many songs, setLists, and artists there are in the database after running this method. This is testing process which is alright, but I think testing the end result (is shit in the db) is probably more sound.